camelot/core/backend.py

```diff
@@ -94,26 +97,36 @@ def __init__(self):
         backend.action_runner().onConnected()
 
     @classmethod
-    def _execute_serialized_request(cls, serialized_request, response_handler):
-        try:
-            response_handler.send_response(Busy(True))
-            AbstractRequest.handle_request(
-                serialized_request, response_handler, response_handler
-            )
-            response_handler.send_response(Busy(False))
-        except Exception as e:
-            LOGGER.error('Unhandled exception in model process', exc_info=e)
-            import traceback
-            traceback.print_exc()
-        except SystemExit:
-            LOGGER.debug('Terminating')
-            raise
-        except:
-            LOGGER.error('Unhandled event in model process')
+    def _enqueue_serialized_request(cls, serialized_request, response_handler):
+        print('enqueue:', serialized_request)
+        request = json.loads(serialized_request)
+        cls._queue.append((request, response_handler))
+        cls._process_queue()
```

Comment:
Zonder deze call komt de cancel ook in de queue op het juiste moment. De actions werken dan wel niet.

Met deze call komt de cancel pas in de queue na het afhandelen van de eerste actie.